html/cross-origin-opener-policy/resources/common.js

```diff
@@ -92,4 +92,16 @@ function run_coop_test_iframe (documentTitle, iframe_origin, popup_origin, popup
       });
       document.body.append(frame);
   }, `${documentTitle} with ${iframe_origin.name} iframe opening popup a ${popup_origin.name} with COOP: ${popup_coop}`);
-}
\ No newline at end of file
+}
+
+// Need to wait until the page is fully loaded before navigating
+// so that it creates a history entry properly.
+const fullyLoaded = new Promise((resolve, reject) => {
+  addEventListener('load', () => {
+    requestAnimationFrame(() => {
+      requestAnimationFrame(() => {
+        resolve();
+      });
+    });
+  });
+});
```

Comment:
It seems somewhat bad to start running this everywhere if we only use it in two places.

Reply:
Fixed in 8db3e7b